allennlp/training/model.py

```diff
@@ -0,0 +1,89 @@
+from typing import Any, List, Dict, Union
+import torch
+
+
+class Model(torch.nn.Module):
+    """
+    This abstract class represents a model to be trained. Rather than relying on the
+    raw torch Module, we provide a slightly different API.
+    """
+
+    def forward(self,
```

Comment:
Somehow github dropped my previous comment on this: I think it makes more sense to separate the loss computation from `forward` - it's pretty confusing to have this sometimes return the loss, and sometimes not.  What's the motivation for doing it this way?  Because the targets are included in `inputs`?  If that's the only reason, I think it'd be better to solve it by separating them in the data API somehow, instead of doing it this way.